File: app/src/main/java/com/bq/zowi/models/commands/ForwardBackwardCommand.java
package com.bq.zowi.models.commands;

import com.bq.zowi.models.commands.Command;

/* JADX INFO: loaded from: classes.dex */
public class ForwardBackwardCommand extends MovementCommand {
    private static final Command.Action[] FORWARD_BACKWARD_ACTIONS = {Command.Action.WALK, Command.Action.FLAPPING};
    private static final Command.Direction[] FORWARD_BACKWARD_DIRECTIONS = {Command.Direction.FORWARD, Command.Direction.BACKWARD};

    public ForwardBackwardCommand(Command.Action action, Command.Direction direction, long duration) {
        setAction(action);
        setDirection(direction);
        setDuration(duration);
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command.Action[] getAllowedActions() {
        return FORWARD_BACKWARD_ACTIONS;
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command.Direction[] getAllowedDirections() {
        return FORWARD_BACKWARD_DIRECTIONS;
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command copy() {
        return new ForwardBackwardCommand(getAction(), getDirection(), getDuration().longValue());
    }
}

File: zowi-core/src/main/kotlin/com/bq/zowi/models/commands/ForwardBackwardCommand.kt
package com.bq.zowi.models.commands

class ForwardBackwardCommand(
    action: Command.Action,
    direction: Command.Direction,
    duration: Long
) : MovementCommand() {

    init {
        setAction(action)
        setDirection(direction)
        setDuration(duration)
    }

    override fun getAllowedActions(): Array<Command.Action> = FORWARD_BACKWARD_ACTIONS
    override fun getAllowedDirections(): Array<Command.Direction> = FORWARD_BACKWARD_DIRECTIONS
    override fun copy(): Command = ForwardBackwardCommand(getAction(), getDirection()!!, getDuration()!!)

    companion object {
        private val FORWARD_BACKWARD_ACTIONS = arrayOf(Command.Action.WALK, Command.Action.FLAPPING)
        private val FORWARD_BACKWARD_DIRECTIONS = arrayOf(Command.Direction.FORWARD, Command.Direction.BACKWARD)
    }
}
